File: demos/node/sayHello.mjs
import 'dotenv/config';

const CLIENT_ID = process.env.CLIENT_ID;
const CLIENT_SECRET = process.env.CLIENT_SECRET;

async function getAccessToken(id,secret) {

	const params = new URLSearchParams();
	params.append('client_secret', secret);
	params.append('grant_type', 'client_credentials');
	params.append('scope', 'openid,AdobeID,read_organizations');

	let resp = await fetch(`https://ims-na1.adobelogin.com/ims/token/v2?client_id=${id}`, 
		{ 
			method: 'POST', 
			body: params
		}
	);

	let data = await resp.json();
	return data.access_token;

}

(async () => {

	let token = await getAccessToken(CLIENT_ID, CLIENT_SECRET);

	let resp = await fetch('https://image.adobe.io/pie/psdService/hello', {
		headers: {
			'Authorization':`Bearer ${token}`,
			'x-api-key': CLIENT_ID
		}
	});

	console.log(await resp.text());

})()

File: demos/python/sayHello.py
import os 
import requests 

CLIENT_ID = os.environ.get('CLIENT_ID')
CLIENT_SECRET = os.environ.get('CLIENT_SECRET')

def getAccessToken(id, secret):

	response = requests.post(f"https://ims-na1.adobelogin.com/ims/token/v2?client_id={id}&client_secret={secret}&grant_type=client_credentials&scope=openid,AdobeID")
	return response.json()

def sayHello(id, token):
	response = requests.get(f"https://image.adobe.io/pie/psdService/hello", headers = {"Authorization": f"Bearer {token}", "x-api-key": id })
	return response.text

token = getAccessToken(CLIENT_ID, CLIENT_SECRET)['access_token']

response = sayHello(CLIENT_ID, token)
print(f"Response from hello endpoint: {response}")

